yarn-project/simulator/src/acvm/oracle/typed_oracle.ts

```diff
@@ -60,14 +45,15 @@ class OracleMethodNotAvailableError extends Error {
   }
 }
 
+const USE_ORACLE_DETERMINISM = !!process.env.SEED;
 /**
  * Oracle with typed parameters and typed return values.
  * Methods that require read and/or write will have to be implemented based on the context (public, private, or view)
  * and are unavailable by default.
  */
 export abstract class TypedOracle {
   getRandomField(): Fr {
-    return Fr.random();
+    return USE_ORACLE_DETERMINISM ? Fr.ONE : Fr.random();
```

Comment:
Why not use the RandomnessSingleton here? An implicit assumption of getRandomField is that you'd get different values if you called it different times, which is broken by returning the same constant. 

Though I can see the appeal of it, for sure...
![dilbert](https://github.com/AztecProtocol/aztec-packages/assets/429604/aa3e1917-cf41-4e68-b359-f2a568ada9ad)

Reply:
If this used the randomness singleton with a unique string per noir execution and was called in a deterministic order, then it could be used. That'd be ideal and I'll think about it